src/context/src/rendering_context.rs

```diff
@@ -0,0 +1,90 @@
+extern crate amethyst_renderer;
+extern crate glutin;
+extern crate gfx_window_glutin;
+extern crate gfx_device_gl;
+extern crate gfx;
+
+use amethyst_config::Element;
+use std::path::Path;
+use self::glutin::{Window, WindowBuilder};
+use self::amethyst_renderer::Renderer;
+use self::amethyst_renderer::target::{ColorFormat, DepthFormat};
+use self::gfx_device_gl::{Device, Factory, Resources, CommandBuffer};
+use self::gfx::handle::{RenderTargetView, DepthStencilView};
+
+config!(
+    /// Contains display config,
+    /// it is required to create a `(Window, RenderingContext)` pair
+    struct DisplayConfig {
+        pub title: String = "Amethyst game".to_string(),
+        pub fullscreen: bool = false,
+        pub dimensions: Option<(u32, u32)> = None,
```

Comment:
I think it should be an `Option` for ergonomics reasons. If `display_config.dimensions` is an `Option` then it is possible to get a `Window` in fullscreen with native resolution just by not passing anything to `display_config.dimensions` field, and if it is not in fullscreen then it will just default to some arbituary dimensions.

If `display_config.dimensions` is not an `Option` then in fullscreen resolution would be set to some hard coded value that doesn't necessarily match the native resolution in case if the proper resolution is not set in the config. Which might be unexpected.